Change Log
===========

MIP v3.0 --> v4.0

mip.pl

- Fixed chrY for female in SVRanking
- Fixed bug in SambambaDepth causing logging to be turned off
- Fixed bug causing MostCompleteBAM being incorrectly added when launching single module
- Fixed bug causing info for MarkDuplicates to not be logged to qc_sampleInfo
- Added flag for creating index file with GatherBAMFile. Default to TRUE
- Temporary fix allowing samtools to create index for GATKBaseRecalibration BAM instead of Picard to accomodate pileup.js
- Modified javaUseLargePages and reduceIO conditional statement to be implicitly boolean
- Added creation of directory for analysis config file unless directory already exists
- Validate all parameters
- Changed percentag_mapped_reads to percentage_mapped_reads
- Added Vcftools and Plink2 versions to qcmetrics
- Changed default for GATKReAligner to “0”
- Added SLURM QOS to sbatch header
	 - Allowed values: [low, normal, high]. Defaults to "normal”.
- Add program processing Markduplicates metric to qc_sampleInfo as “ProcessedBy” for log purpose
- Remade array parameter inFilesDirs to hash parameter inFilesDir
- Added CLNREVSTAT to config
- Changed analysisType from scalar to hash (-at sampleID=analysisType)
	- Default genomes
	- Can be supplied from pedigree using "Sequence_type"
	- Enables performing analysis under correct fastq location and with correct parameters for sample specific analyses
	- Changed default output structure
        - FamilyID is now root
        - Data is stored under root in analysisType dir
        - pedigree is stored under root
        - scripts, mip_log, config and qc_sampleInfo are stored under FamilyID/analysis
- Changed "exomes" and "genomes" to "wes" and "wgs" respectively
-  Added new baitset shortcut in pedigree: Agilent_SureSelectFocusedExome.V1.GRCh37_targets.bed
- Modified UpdateToAbsolutePath to get info from definitions directly
- Move all rio BAMProcessing to same in and out directory to enable skipping of programs
	- Modules that variant callers and qc modules output under their own dir, but gathered data is processed under alignment directory
- Call sub in RankVariants for select file and adjust MT|M
- Changed bcf generation to vcf.gz generation + tabix index
- Changed rankedBCFFile to rankedBinaryFile
- Changed svRankedBCFFile to svRankedBinaryFile
- Changed tag in qc_SampleInfo from BCFFile to VCFBinaryFile and VCFSVBCFFile to SVBinaryFile
- Change to tabix from bcf since it seems more forgiving and produce similar compression level
- Added “—rank_results” in genmod score for producing rank view in Scout
- Remove rank score sorting of clinical/ research
- Add bwa log and HLA log to qc_sampleInfo and copying back to hds
- Added decomposing using bcftools for variant callers and normalization for GATKVariantRecalibration
- GRCH38 Error: Error: Field name 'phyloP46way_primate' not found dbSnpEff
	- phyloP46way_primate -> phyloP20way_mammalian
	- phyloP100way_vertebrate -> phyloP7way_vertebrate
	- phastCons46way_primate ->  phastCons20way_mammalian
	- phastCons100way_vertebrate -> phastCons7way_vertebrate
- Added novel calculation of F-score using plink2
- Add CLNVAR curation status (CLNREVSTAT) into rank model and bumped rank_model to version 1.17
- Added —disable_auto_index_creation_and_locking_when_reading_rods to GATKReAlign and GATKBaserecal
- Added SWEREF into rank model and to default (NOTE: only car 22 pending actual release)
- Added ‘variant integrity’ to sampleCheck
- Modifed InbreedingFactor regExp for plink2 .het output
- Added module to split fastq files, move original files to sub dir and then exit
- Exchanged 'vars' for our
- Made Path and outdataDirection and outDataFile be collected by recursive strategy
- Let MIP detect if no affected is in pedigree and warn and turn of genmod models, score and compound
- Updated VTref switch to avoid modifying same reference twice
- Updated rank model to locate SIFT and PolyPhen from CSQ-field directly
- Added LoFtool gene intolerance prediction to rank model
- Updated rank_model to version 1.18
- Added additional sorting of SV variants after vcfanno annotation
- Added extra sort of SV variants after ranking
- Process SV exome|rapid as one file instead of splitted per contig to ensure that no contig will lack variants
- Added insert qc metrics to qc_sampleInfo and qcmetrics
- Added support for interleaved fastq files and relaxed file convention criteria
	- Interleaved info is gathered from fastq header for read direction 1
	- BWAMem alignment is automatically adjusted accordingly
	- MIP now supports mixing of SE, PE and PE-interleaved files within the same fastq directory
	- Relaxed file convention criteria by collecting mandatory info from fastq header - just require sampleID in filename
	- Add fake date since this information is not recorded in fastq header for standardised file spec
- Added support for metadata in yaml format (pedigree and other meta data)
- MIP will look at the filending to detect file format
- Set mandatory keys in Plink pedigree format to be lower case throughout in MIP output
- Added additional reformating of yaml value for "sex" and "phenotype" under new keys to adhere to Plink format when required
- Modified and added pedigree templates
- Removed instanceTag and researchEthicalApproved
- Added temp_dir to genmod annotate and filter in sub VT
- Modfied pedigree file to genmod calls to use famFile and changed default to 'ped'
- Modified pedigree file default ending to “.yaml”
- Modified qcCollect to new yaml structure
    - Added sampleID level for evaluate
    - Cleaned up code
- Added cut-offs for evaluation of mendel and father
- Added collection of expected_coverage from ped.yaml and relay to qcCollect for evaluation
- Removed extra feature annotations and some VEp field parsing
    svVcfParserRangeFeatureAnnotationColumns:
      - 3 = Ensembl_gene_id - REMOVED
      - 4 = HGNC_symbol
      - 5 = Phenotypic_disease_model - REMOVED
      - 6 = OMIM_morbid - REMOVED
      - 7 = Ensembl_transcript_to_refseq_transcript - REMOVED
      - 8 = Gene_description - REMOVED

    svVcfParserSelectFeatureAnnotationColumns:
      - 3 = HGNC_symbol
      - 10 = Phenotypic_disease_model - REMOVED
      - 11 = OMIM_morbid - REMOVED
      - 14 = Ensembl_gene_id - REMOVED
      - 16 = Reduced_penetrance - REMOVED
      - 17 = Clinical_db_gene_annotation - REMOVED
      - 18 = Disease_associated_transcript - REMOVED
      - 19 = Ensembl_transcript_to_refseq_transcript - REMOVED
      - 20 = Gene_description - REMOVED
      - 21 = Genetic_disease_model - REMOVED

    - Removed additional VEP parsing:
      - GeneticRegionAnnotation
      - HGVScp
      - INTRON
      - EXON
      - STRAND
      - HGVSc
      - HGVSp
- Added GBCF file creation and key-path to qc_sampleInfo
- Added pedigree_minimal (.fam file) and config_file_analysis to qc_sample_info
- Add test of SV files in analysisrunstatus
- Expect select file to have full path and not be located in MIP reference directory
- Moved sacct module to case level

Install.pl

- Added boolean flag condaUpDate and changed flag perlInstall to boolean
- Renamed preferBioConda to preferShell and made it boolean
- Renamed flag update to noUpdate and made it boolean
- Activated CNVnator installation 
- Added install script to conda env for printing software versions connected to MIP version
- Added Validate parameter checks, named arguments and sub description
- Updated genmod to version 3.5.6
- Added ability to set python version when creating conda env
- Updated chanjo to v4.0.0

vcfParser.pl

- Removed Sift and Polyhen parsing from CSQ field
- Change SYMBOL to HGNC_ID in vcfparser
- Added per_gene option

qcCollect.pl

- Changed percentag_mapped_reads to percentage_mapped_reads
- Added raw total sequences and reads mapped to qcCollect
- Added Vcftools and Plink2 versions to qcmetrics
- Updated regExp file to version 12

MIP v2.6 --> v3.0

- Added Net/SSLeay.pm to install.pl
- Added option to skip perl install
- Added Manta, Delly, FT and CNVnator as structural variant callers
- Added modules CombineStructuralVariants, SVVariantEffectPredictor, SVVCFParser, SVRanking
- Added merging of samples in “other” chains to family chain for parallel modules
- Added CNVnator version. Had to be done at start up since CNVnator does not add its version to the output.
- Added Delly version on sample level
- Added Manta version  on sample level
- Added SVVEP version and cache version 
- Fixed bug causing VEP version to be lost for snvs/indels
- Added SVVCFParser version
- Added SVGenmod/rankModel version
- Added test for GATKCombineVariantsPrioritizeCaller to not include turned of variant callers
- Added snpEff download of reference genome to avoid race conditions
- Fixed python virtuelenvironment to not check programs if uve = 0
- Added VEP/SVVEP assembly, gencode, gene build, HGMDPublic, polyphen, regbuild, Sift, version to qcmetrics
- made NIST ID settable
- Removed PicardMergeSwitch, now all files are merged or renamed (single files) for more consequent naming and easier processing
- Renamed ‘fileEnding’ to ‘fileTag’ and ‘removefileEnding’ to ‘fileEnding’ 
- Change name of BAMCalibrationAndGTBlock to only BAMCalibrationBlock
- List::Util is in core module perl 5.18 replaces List::MoreUtils
- Use say instead of print where relevant
- Use internal Perl system commands instead of UNIX (copy, make_path)
- Removed mip log file if present in config to avoid appending to old log file. Supply on log file on cmd if you want to append to log file.
- Added plink2 installation via bioconda in install script
- Changed binary i MIP from plink to plink2 
- Added MultiQC in install script and as MIP module
- Changed samtools stats module to include complete report for MultiQC processing
- Made pPicardToolsMergeSamFiles mandatory: Always run even for single samples to rename them correctly for standardised downstream processing. Will also split alignment per contig and copy to temporary directory for '-rio 1' block to enable selective removal of block submodules.
- Added LOFTEE VEP plugin: https://github.com/konradjk/loftee
- Added LofTool VEP plugin
- Added Modern::Perl ‘2014’
- Added PERL_UNICODE=SAD to install script, and hence bash_profile - stdin, stdout, and stderr to UTF‑8 as well as @ARGV and data handlers
- Use UTF-8 for all source script
- Added encoding UTF-8 pragma for open to default expect unicode when opening and writing
- Enforce perl 5.18 version
- Added autodie for generalised error and exception handling
- Removed dateTime and use less cumbersome core module Time::Piece 
- Removed DV and added AD for samtools mpileup
- Added joint calling of SV using Manta
- Added SV analysis of exomes using Manta
- Modified CombineSVVariants to use Delly and CNVnator on sample-level and Manta on family level
- Added bcf generation of ranked vcf both select file and research
- Fixed bug in temp directory
- Bumped install version to 3.5.1
- Added Genmod temp dir flag
- Added sacct commands to trap for each sbatch to relay progress to MIP log file.status
- Made Sacct dependency into afterany
- Added pPrepareForVariantAnnotationBlock
- Removed pythonVirtualEnv and commands as conda is prefered
- Added sourceEnvironmentCommand
- Added '-pp' and '-ppm'
- Add bcf conversion of select and research variants to MIP
- Added check of programs mode to allowed values, more strict parsing for flaggs expecting numbers
- Select variants prior to Plink processing using GATK Select variants
	- Move processing to node, but keept final output printing to hds
- Added SV annotation using 1000G SV and vcfanno -ends
	- Added vcfanno, lua, config
	- Annotate from 1000G SV
 	- Modified svrank_mdodel to take 1000G frequency in account
- Add vcfanno version in SVCombineVariantCallSets
- Updated fastqc to version 0.11.5
- Updated bwa to version 0.7.13
- Updated sambamba to version 0.6.1
	- Added “—fix-mate-overlaps” to avoid counting overlapping reads twice
	- Removed Sambamba version from MIP flagg
- Updated picardtools to version 2.3.0
- Updated Chanjo to version 3.4.1
- Updated Manta to version 0.29.6
- Updated Genmod to version 3.5.2
- Updated MultiQC to version 0.6
- Updated Vip to version 84
- Added Picardtools Markduplicates as a option and default
- Added more SambambaMarkDup options
- Make sambamba flagstats into subroutine to be used for all markdup
- Remade capture kit options into 1 hash flag, which will build all associated files if 1 is lacking 
- Make  Covariates to be used in the recalibration in GATKBASERCAL to be flag and array option
	- annotations, -Know and -knownSites 
- Remade VEP install assembly flag to be array and used rerun install for each assembly version
- Remade SnpEff install genomeVersion flag to be array and used rerun download for each genome version
- Added assembly flag to VEP script and alias it to use GRCh prefix and number 
- Fixed chr prefix for chanjo sex check
- Updated to GATK version 3.6
- Created contig splitted target files on the fly for non genome analysis to reduce the running time of GATK Realign, BaseRecal and Haplotype
- Added sub ReplaceIUPAC and used it on freebayes and samtools mpileup vcfs
- Changed analysisType default from exomes to genomes

MIP v2.4 --> v2.6

- Updated GATK to 3.5
- Added static binning capability for base recalibration (BQSR)
- Added option --disable_indel_quals to BSQR
- Added limit for exomes to only use target bases in recalibration
- Added MTAF to SnpEff and vcfParser for MT frequency annotations
- Added 'trio' detection to parameters instead of scriptParameters to avoid writing key to config
- Fixed bug when supplying -sambambaDepthCutOffs on cmd
- MIP now handles updating to absolute path for  comma separated parameters correctly
- Removed write to cmd string in mip log for some internal parameters
- Updated install script
	- Added PIP to the condo env upon creation
	- Add check that condo is executable in system before launching rest of installation
	- Install script can now detect existing condo env and change cmd to accommodate 
	- Added sambamba (0.5.9), vt (2015.11.10), bedtools (2.25.0), htslib (1.2.1) to bioconda install
	- Added option to prefer Bioconda install over shell for overlapping modules
	- Added soft link creation sub routine
	- Use soft link sub for sambamba (both bioconda and Shell) 
	- Add soft link to snpEff och SnpSift for bioconda install
- Update FASTQC to 11.4 via bioconda
- Updated SnpEff to v4_2 via Shell
- Updated Plink to v1.90b3.26 64-bit (26 Nov 2015) via shell
- Updated vcfTools to 0.1.14 via SHELL
- Updated Chanjo to 3.1.1 via PIP
- Updated Genmod to 3.4 via PIP
- Updated Picardtools to 1.141 via bioconda
- Updated Samtools to 1.3
- Updated bcfTools to 1.3
- Updated htslib to 1.3
- Added picardTools installation via SHELL
- Updated VEP to 83 via SHELL
	- Trouble with distribution - htslib and sereal (only issues with testing and not with actual running the script)
	- Added installation of VEP plugin UpDownDistance
- Added use of VEP plugin UpDownDistance for MT contig only to avoid over annotation of the compact MT genome
- Added padding to 10 nucleotides for MT in Vcfparser
- Added test for undetermined in fastq file name and adjust qc-test to skip entirely for these reads
- Added samtools mpileup 
- Added GATKCombineVariants to combine variants calls from multiple variant callers
- Added generalisation for supporting multiple variant callers in MIP dependencies and GATKCombinaVariants
- Added no-fail to sample check
- Modified installation of picardTools and SnpEff
- Add filtering to variant calls from samtools mpileup
- Add samtools/bcfTools versions 
- Add removal of samtools pileup files
- Added test::Harness for TAP summary results and future inclusion of additional test scripts
- Add option to determine priority in variant callers as comma sep string
- Add check of variant callers active compared to prioritise flag
- Add sanity check of prioritisation flag
- Add option to turn on or off installation of programs in install.pl
- Added bcf file compression and indexing as sub
- Added vcfTobcf sub to GATKCombineVariants
- Switched vcf ready file from GATKVariantRecalibration to GATKCombineVariants
- Added Freebayes variant caller
	- Added to removeRedundantFiles
	- Added Freebayes version to qcCollect
- RemoveRedundant files info is now recorded in definition.yaml
- Added GATKCombineVariants to removeRedundant files
- Add bcftools norm to samtools pileup and freebayes output
- Add lastlogFilePath to qc_sampleInfo
- Made lanes and readDirections info more nested
- Add 1000G Phase 3 and Exac to Genmod annotation
- Changed regEx in test.t  to include all until “,” for INFO fields in Header
- Modified bioconda softlinks sub call to only execute if programs are installed
- Added MT.codon table sub for snpEff config to install script
- Remake GENMOD CADD file option to array
- Added padded target intervals to exome analysis again for GATKRealign and GATKHaplotypeCaller
- Reactivate GATKPaddedTarget parameter
- Made associatedPrograms arg into an array instead of a comma sep string
- Fixed check for when a capture kits is lacking from input and fallback to using “latest”
- Remade CheckParameterFiles to work with DataType
- Add evaluation with NIST as a module in MIP
- Fix the . mip.sh to bash mip.sh
- Added reference to define/definitions
- CheckParameterFiles now works with parameterExistsCheck directly instead of “d” and “f” enabling merge of directory and file sections
- Changed if for intervalListFile to be if($IntervalList ) instead of analysisTypeExome|rapid
- Add programType=Aligner to define/definitions 
- Remade sanity check of aligner to count if more than 1 aligner has been switched on (MosaikAligner, BWASampe, BWAMEM)
- Dynamic setting of ‘aligner’ depending in aligner supplied by outDirectoryName
- Renamed aligner to alignerOutDir
- Added genmod max_af
- Added canonical to VEP features

MIP v2.0 --> v2.4

- Bugfixes
- Updated most program version (see docs) and databases
	- Logs versions and databases
- Added -pVT
- Added -allSites option to GenoTypeVCFs
- Added version tag to definitions.yaml
- Cleaned some old parameter names
- Added test for parameter compatibility between defineParameters.yaml and config
- Added new parameter snpSiftAnnotationOutInfoKey
- Changed SnpSift_ for 1000G and EXACAF to facilitate downstream processing since both work on KEY=AF
- Remade dbsnpAF parsing to accommodate multiple entries for the same env
- Added vt decompose and normalise subroutine for both reference and variant vcf
- Removed vcf_parser —split
- MIP now works only on config tags from select file meta data header for select genes
- Added genmod version and removed RankVariants version
- Add test for VEP cache and directory version linked
- Added option OverclippedReadFilter to GATKBaseRecalibration/PrintReads
- Exchange grep for any in array check and use eq instead of // for stringency
- Added vt decompose and normalise subroutine call for relevant downloadable references ("indels", "mills", "dbsnp", "hapmap", "dbsnpex", "1000g_snps")
- Add check for ingoing references that VT has been used if VT is on
- Fixed bug in AnalysisRunStatus modules caused when first processing -rio 1 and then -rio 0
- Fixed bug when adding samples to pedigree to already processed samples
- Removed Radial:sw and LR_score from dbNSFP annotation as these have become obsolete
- Remade RemoveRedundant files
- Added bcf compression alternativ
- Added perl oneliner to VT that removes '*' alt.allele after decomposing as it does not add any new info


MIP v1.0 --> v2.0

- Major code refactoring
- Bugfixes
- Updated most program version (see docs) and databases
	- Logs versions and databases
- Removed modules -pMerge_anvar, -pAdd_dp
	- MIP no longer creates master templates, instead this is taken care of dynamically
- Added -pVeP, -pSnE, -pVcP -pChanjoSexCheck
- Module PicardSortSam is now integrated in alignment modules
- Use VCF format where appropriate
	- Created standardised VCF list levels (",", ":", "|")
- Clinical transcripts are selected after VEP annotation using VCFParser
	- Removes ethical issue with overlapping genes
- Full resolution in annotation
	- Gene
	- Transcripts
	- Multiple alleles
	- Split multi allelic calls into single records
	- Use SO terms
	- Calculate Sift an PolyPhen per transcript and allele
	- Remade transcript and cDNA and protein info from VEP CSQ field
	- Switched from MAF to AF
- Use Log4Perl for logging
- All processes create temp directory on (default @nodes)
- Creates automatic migration to and from nodes
- Deploy more aggressive scatter/gather technique. Processing per contig whenever possible.
- Analyse order in contig size not number
- Use piping in SnpSift annotation and where possible 
- Reduce IO between nodes using -rio flag. Will run modules sequentially where appropriate.
	- Created automatic removal of files when appropiate at tempDir

* Flag changes
	- -huref/--humanGenomeReference --> -hgr/--humanGenomeReference
	- -rea/--researchEthicalApproval Tag for displaying research candidates in Scout (defaults to "notApproved")
	- -tmd/--tempDirectory Set the temporary directory for all programs (defaults to "/scratch/SLURM_JOB_ID";supply whole path)
	- -nrm/--nodeRamMemory The RAM memory size of the node(s) in GigaBytes (Defaults to 24)
	- -ges/--genomicSet Selection of relevant regions post alignment (Format=sorted BED; defaults to "")
	- -rio/--reduceIO Run consecutive models  at nodes (defaults to "0")
	- -l/--logFile Mip log file (defaults to "{outDataDir}/{familyID}/mip_log/{timestamp}/{scriptname}_{timestamp}.log")
	- -pGZ/--pGZip --> -pGZ/--pGZipFastq
	- -pFQC/--pFastQC --> -pFqC/--pFastQC 
	- -moaannpe/--mosaikAlignNeuralNetworkPeFile --> -moaape/--mosaikAlignNeuralNetworkPeFile
	- -moaannse/--mosaikAlignNeuralNetworkSeFile --> -moaase/--mosaikAlignNeuralNetworkSeFile
	- -pBWA_mem/--pBwaMem --> -pMem/--pBwaMem
	- -bwamemrdb/--bwaMemRapidDb --> -memrdb/--bwaMemRapidDb
	- -pBWA_aln/--pBwaAln --> -pAln/--pBwaAln 
	- -pBWA_sampe/--pBwaSamp --> -pSap/--pBwaSampe
	- -picardpath/--picardToolsPath --> -ptp/--picardToolsPath
	- -picttmpd/--PicardToolsTempDirectory --> removed
	- -pPicT_sort/--pPicardToolsSortSam  --> removed
	- -pPicT_merge/--pPicardToolsMergeSamFiles --> -pPtM/--pPicardToolsMergeSamFiles
	- -pPicT_mergerr/--pPicardToolsMergeRapidReads  -> -pPtMR/--pPicardToolsMergeRapidReads
	- -picT_mergeprev/--picardToolsMergeSamFilesPrevious --> -ptmp/--picardToolsMergeSamFilesPrevious
	- -pPicT_markdup/--pPicardToolsMarkduplicates --> -pPtMD/--pPicardToolsMarkduplicatesWithMateCigar
	- -pCh_B/--pChanjoBuild --> -pChB/--pChanjoBuild
	- -pChS/--pChanjoSexCheck
	- -pCh_C/--pChanjoCalculate --> -pChA/--pChanjoAnnotate
	- -chccut/--chanjoCalculateCutoff --> -chacut/--chanjoAnnotateCutoff
	- -pCh_I/--pChanjoImport --> -pChI/--pChanjoImport
	- -pCC_bedgc/--pGenomeCoverageBED --> -pGcB/--pGenomeCoverageBED
	- -xcov/--xCoverage --> -gcbcov/--GenomeCoverageBEDMaxCoverage
	- -pCC_picmm/--pPicardToolsCollectMultipleMetrics --> -pPtCMM/--pPicardToolsCollectMultipleMetrics
	- -pCCE_pichs/--pPicardToolsCalculateHSMetrics --> -pPtCHS/--pPicardToolsCalculateHSMetrics
	- -extbl/--exomeTargetBedInfileLists --> -ptchsetl/--exomeTargetBedInfileLists
	- -extpbl/--exomeTargetPaddedBedInfileLists --> -ptchsetpl/--exomeTargetPaddedBedInfileLists
	- -pRCP/--pRCovPlots --> -pRcP/--pRCovPlots
	- -gatkpath/--genomeAnalysisToolKitPath --> -gtp/--genomeAnalysisToolKitPath
	- -gatkbdv/--GATKBundleDownLoadVersion --> -gbdv/--GATKBundleDownLoadVersion
	- -gatktmpd/--GATKTempDirectory --> removed
	- -gatktpbl/--GATKTargetPaddedBedIntervalLists --> -gtpl/--GATKTargetPaddedBedIntervalLists
	- -gatkdcov/--GATKDownSampleToCoverage --> -gdco/--GATKDownSampleToCoverage
	- -pGATK_real/--pGATKRealigner  -->  -pGrA/--pGATKRealigner 
	- -gatkrealknset1/--GATKReAlignerINDELKnownSet1 --> -graks1/--GATKReAlignerINDELKnownSet1
	- -gatkrealknset2/--GATKReAlignerINDELKnownSet2 --> -graks2/--GATKReAlignerINDELKnownSet2
	- -pGATK_baserecal/--pGATKBaseRecalibration --> -pGbR/--pGATKBaseRecalibration
	- -gatkbaserecalknset/--GATKBaseReCalibrationSNPKnownSet --> -gbrkse/--GATKBaseReCalibrationSNPKnownSet
	- -pGATK_hapcall/--pGATKHaploTypeCaller --> -pGhC/--pGATKHaploTypeCaller
	- -gatkhapcallsnpknset/--GATKHaploTypeCallerSNPKnownSet --> -ghckse/--GATKHaploTypeCallerSNPKnownSet
	- -pGATK_genotype/--pGATKGenoTypeGVCFs --> -pGgT/--pGATKGenoTypeGVCFs
	- -gatkgenotyperefgvcfinfile/--GATKGenoTypeGVCFsRefGVCFInfile --> -ggtgrl/--GATKGenoTypeGVCFsRefGVCF
	- -pGATK_varrecal/--pGATKVariantRecalibration  --> -pGvR/--pGATKVariantRecalibration
	- -gatkexrefsnp/--GATKExomeReferenceSNPs --> -gvrtss/--GATKVariantReCalibrationTrainingSetDbSNP
	- -gatkvarrecaltrhapmap/--GATKVariantReCalibrationTrainingSetHapMap --> -gvrtsh/--GATKVariantReCalibrationTrainingSetHapMap
	- -gatkvarrecaltrd1000Gsnp/--GATKVariantReCalibrationTrainingSet1000GSNP --> -gvrtsg/--GATKVariantReCalibrationTrainingSet1000GSNP
	- -gatkvarrecaltromni/--GATKVariantReCalibrationTrainingSet1000GOmni --> -gvrtso/--GATKVariantReCalibrationTrainingSet1000GOmni
	- -gatkvarrecaltrdbmills/--GATKVariantReCalibrationTrainingSetMills -->  -gvrtsm/--GATKVariantReCalibrationTrainingSetMills 
	- -gatkvarrecaltsfilterlevel/--GATKVariantReCalibrationTSFilterLevel --> -gvrtsf/--GATKVariantReCalibrationTSFilterLevel 
	- -gvrevf/--GATKVariantReCalibrationexcludeNonVariantsFile
	- -gvrsmr/--GATKVariantReCalibrationSpliMultiRecord
	- -pGATK_phaseTr/--pGATKPhaseByTransmission --> -pGpT/--pGATKPhaseByTransmission
	- -pGATK_readPh/--pGATKReadBackedPhasing --> -pGrP/--pGATKReadBackedPhasing
	- -gatkreadphphaseqthr/--GATKReadBackedPhasingPhaseQualityThresh --> -grpqth/--GATKReadBackedPhasingPhaseQualityThreshold
	- -pGATK_varevalall/--pGATKVariantEvalAll --> -pGvEA/--pGATKVariantEvalAll
	- -pGATK_varevalexome/--pGATKVariantEvalExome --> -pGvEE/--pGATKVariantEvalExome
	- -gatkvarevaldbsnp/--GATKVariantEvalDbSNP --> -gveedbs/--GATKVariantEvalDbSNP
	- -gatkvarevaldbgold/--GATKVariantEvalGold --> -gveedbg/--GATKVariantEvalGold
	- -pANVAR/--pAnnovar --> -pAnV/--pAnnovar
	- -anvarpath/--annovarPath --> -anvp/--annovarPath
	- -anvargbv/--annovarGenomeBuildVersion --> -anvgbv/--annovarGenomeBuildVersion
	- -anvartn/--annovarTableNames --> -anvtn/--annovarTableNames 
	- -anvarstn/--annovarSupportedTableNames --> -anvstn/--annovarSupportedTableNames
	- -anvarmafth/--annovarMAFThreshold --> -anvarmafth/--annovarMAFThreshold
	- -pVeP/--pVariantEffectPredictor Annotate variants using VEP (defaults to "1" (=yes))
	- -vepp/--vepDirectoryPath Path to VEP script directory (defaults to ""; supply whole path)
	- -vepc/vepDirectoryCache Specify the cache directory to use (supply whole path, defaults to "")
	- -vepf/--vepFeatures VEP features (defaults to ("refseq","hgvs","symbol","numbers","sift","polyphen","humdiv"); comma sep)
	- -pVcP/--pVCFParser Parse variants using vcfParser.pl (defaults to "1" (=yes))
	- -vcpvt/--vcfParserVepTranscripts Parse VEP transcript specific entries (defaults to "0" (=no))
	- -vcprff/--vcfParserRangeFeatureFile Range annotations file (defaults to ""; tab-sep)
	- -vcprfa/--vcfParserRangeFeatureAnnotationColumns Range annotations feature columns (defaults to ""; comma sep)
	- -vcpsf/--vcfParserSelectFile File containging list of genes to analyse seperately (defaults to "";tab-sep file and HGNC Symbol required)
	- -vcpsfm/--vcfParserSelectFileMatchingColumn Position of HGNC Symbol column in SelectFile (defaults to "")
	- -vcpsfa/--vcfParserSelectFeatureAnnotationColumns Feature columns to use in annotation (defaults to ""; comma sep)  
	- -pSnE/--pSnpEff Variant annotation using snpEFF (defaults to "1" (=yes))
	- -snep/--snpEffPath Path to snpEff. Mandatory for use of snpEff (defaults to "")
	- -snesaf/--snpSiftAnnotationFiles Annotation files to use with snpSift (comma sep)
	- -snesdbnsfp/--snpSiftDbNSFPFile DbNSFP File (defaults to "dbNSFP2.6.txt.gz")
	- -snesdbnsfpa/--snpSiftDbNSFPAnnotations DbNSFP annotations to use with snpSift (defaults to ("SIFT_pred","Polyphen2_HDIV_pred","Polyphen2_HVAR_pred","LRT_pred","MutationTaster_pred","GERP++_NR","GERP++_RS","phastCons100way_vertebrate","1000Gp1_AF","ESP6500_AA_AF"); comma sep)
	- -pRankVar/--pRankVariants --> -pRaV/--pRankVariants
	- -rs/--rankScore --> removed
	- -gf/--geneFile --> -ravgf/--geneFile
	- -imdbfile/--ImportantDbFile Important Db file (Defaults to "") --> removed
	- -imdbte/--ImportantDbTemplate Important Db template file used to create the specific family '-im_dbmf' master file (Defaults to "") --> removed
	- -imdbmf/--ImportantDbMasterFile Important Db master file to be used when selecting variants (defaults to "{outDataDir}/{familyID}/{familyID}.intersectCollect_selectVariants_db_master.txt";Supply whole path) --> removed
	- -imdbfof/--ImportantDbFileOutFiles The file(s) to write to when selecting variants with intersectCollect.pl. Comma sep (defaults to "{outDataDir}/{familyID}/{aligner}/GATK/candidates/ranking/{familyID}_orphan.selectVariants, {outDataDir}/{familyID}/{aligner}/GATK/candidates/ranking/clinical/{familyID}.selectVariants"; Supply whole path/file) --> removed
	- -ravcs/--caddWGSSNVs Annotate whole genome sequencing CADD score (defaults to "0" (=no))
	- -ravcsf/--caddWGSSNVsFile Whole genome sequencing CADD score file (defaults to "whole_genome_SNVs.tsv.gz")
	- -ravc1kg/--cadd1000Genomes 1000 Genome cadd score file (defaults to "0" (=no))
	- -ravc1kgf/--cadd1000GenomesFile 1000 Genome cadd score file (defaults to "1000G.tsv.gz")
	- -ravwg/--wholeGene Allow compound pairs in intronic regions (defaults to "1" (=yes))
	- -ravrm/--rankModelFile Rank model config file (defaults to "")
	- -pSCheck/--pSampleCheck --> -pScK/--pSampleCheck
	- -pQCC/--pQCCollect --> -pQcC/--pQCCollect
	- -QCCsampleinfo/--QCCollectSampleInfoFile --> -qccsi/--QCCollectSampleInfoFile
	- -QCCregexp/--QCCollectRegExpFile --> -qccref/--QCCollectRegExpFile
	- -pREM/--pRemovalRedundantFiles --> -pReM/--pRemoveRedundantFiles
	- -pAR/--pAnalysisRunStatus --> -pArS/--pAnalysisRunStatus